GPflowOpt/acquisition.py

```diff
@@ -168,47 +170,63 @@ def build_acquisition(self, Xcand):
         return normal.cdf(tf.constant(0.0, dtype=float_type), name=self.__class__.__name__)
 
 
-class AcquisitionBinaryOperator(Acquisition):
-    """
-    Base class for implementing binary operators for acquisition functions, for defining joint acquisition functions
-    """
-
-    def __init__(self, lhs, rhs, oper):
-        super(AcquisitionBinaryOperator, self).__init__()
-        assert isinstance(lhs, Acquisition)
-        assert isinstance(rhs, Acquisition)
-        self.lhs = lhs
-        self.rhs = rhs
+class AcquisitionAggregationOperator(Acquisition):
+    def __init__(self, operands, oper):
+        super(AcquisitionAggregationOperator, self).__init__()
+        for operand in operands:
+            assert isinstance(operand, Acquisition)
+        self.operands = ParamList(operands)
         self._oper = oper
 
     @Acquisition.data.getter
     def data(self):
-        lhsX, lhsY = self.lhs.data
-        rhsX, rhsY = self.rhs.data
+        X = self.operands[0].data[0]
+        Ys = map(lambda operand: operand.data[1], self.operands)
+
         if self._tf_mode:
-            return lhsX, tf.concat((lhsY, rhsY), 1)
+            return X, tf.concat(list(Ys), 1)
         else:
-            return lhsX, np.hstack((lhsY, rhsY))
+            return X, np.hstack(Ys)
 
     def set_data(self, X, Y):
-        offset_lhs = self.lhs.set_data(X, Y)
-        offset_rhs = self.rhs.set_data(X, Y[:, offset_lhs:])
-        return offset_lhs + offset_rhs
+        offset = 0
+        for operand in self.operands:
+            offset += operand.set_data(X, Y[:, offset:])
```

Comment:
I'm guessing this needs to be changed for multi-objective criteria? Taking in multiple columns of Y